src/components/Notifications/items/ProjectJoinRequest/index.js

```diff
@@ -0,0 +1,234 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {
+    RequestCoordinator,
+    RequestClient,
+    requestMethods,
+} from '#request';
+import { reverseRoute } from '#rsu/common';
+import SuccessButton from '#rsca/Button/SuccessButton';
+import DangerButton from '#rsca/Button/DangerButton';
+import WarningButton from '#rsca/Button/WarningButton';
+
+import DisplayPicture from '#components/DisplayPicture';
+import {
+    pathNames,
+    iconNames,
+} from '#constants';
+import _ts from '#ts';
+
+import Notification from '../Notification';
+import styles from './styles.scss';
+
+const propTypes = {
+    // eslint-disable-next-line react/forbid-prop-types
+    notification: PropTypes.object,
+
+    // eslint-disable-next-line react/forbid-prop-types
+    projectJoinApproveRequest: PropTypes.object,
+
+    // eslint-disable-next-line react/forbid-prop-types
+    projectJoinRejectRequest: PropTypes.object,
+
+    className: PropTypes.string,
+};
+
+const defaultProps = {
+    notification: {},
+    className: '',
+    projectJoinApproveRequest: {},
+    projectJoinRejectRequest: {},
+};
+
+const REQUEST_PENDING = 'pending';
+const ROLE_DEFAULT = 'normal';
+const ROLE_ADMIN = 'admin';
+
+// TODO: show error message for request failure
+const requests = {
+    projectJoinApproveRequest: {
+        url: ({
+            params: {
+                projectId,
+                requestId,
+            },
+        }) => `/projects/${projectId}/requests/${requestId}/accept/`,
+        method: requestMethods.POST,
+        body: ({ params: { role } }) => ({ role }),
+        isUnique: true,
```

Comment:
Correct me if i am wrong, but wouldn't setting the request to be unique make it so that user if the user click 2 approve buttons at the same time, the first one would abort?

Reply:
Not for this case, `isUnique` is set true here for the parent coordinator to be able to monitor this request as well.

Reply:
Isn't there a group field to do that?
Doesn't stating that the request is unique mean that there will be only one request of the kind?

Reply:
Let's see what @bibekdahal has to say about this

Reply:
The usecase here looks justified.